### PR TITLE
Dining

### PR DESCRIPTION
**Pull request by lilyidi**

### CODE REVIEW

**Comment by adelq** [comment]
Squash the commits so you only have 1, and next time you can use `git commit --amend` to add to the past commit

File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/adapters/DiningAdapter.java [hunk 1]
```diff
@@ -47,24 +51,38 @@ public View getView(int position, View view, ViewGroup parent) {
...
                         public void call(DiningHall newDiningHall) {
                             diningHall.sortMeals(newDiningHall.menus);
+                            holder.infoIcon.setVisibility(View.GONE);
                             if (diningHall.hasMenu()) {
+                                holder.infoIcon.setVisibility(View.GONE);
+                                progressBar.setVisibility(View.GONE);
                                 holder.menuArrow.setVisibility(View.VISIBLE);
                             }
+                            else{
+                                Log.d("no menu call", diningHall.getName());
```
Comment (dlakata) [inline PennMobile/src/main/java/com/pennapps/labs/pennmobile/adapters/DiningAdapter.java]: remove the logging

File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/adapters/DiningAdapter.java [hunk 2]
```diff
@@ -100,9 +118,15 @@ public void call(Throwable throwable) {
         }
 
         if (diningHall.hasMenu()) {
+            holder.infoIcon.setVisibility(View.GONE);
+            progressBar.setVisibility(View.GONE);
             holder.menuArrow.setVisibility(View.VISIBLE);
         }
-
+        else{
+            progressBar.setVisibility(View.INVISIBLE);
```
Comment (dlakata) [inline PennMobile/src/main/java/com/pennapps/labs/pennmobile/adapters/DiningAdapter.java]: why invisible here, and not gone, as above? just curious

File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/adapters/DiningAdapter.java [hunk 3]
```diff
@@ -47,24 +51,38 @@ public View getView(int position, View view, ViewGroup parent) {
...
                         @Override
                         public void call(DiningHall newDiningHall) {
                             diningHall.sortMeals(newDiningHall.menus);
+                            holder.infoIcon.setVisibility(View.GONE);
                             if (diningHall.hasMenu()) {
+                                holder.infoIcon.setVisibility(View.GONE);
+                                progressBar.setVisibility(View.GONE);
                                 holder.menuArrow.setVisibility(View.VISIBLE);
                             }
+                            else{
```
Comment (dlakata) [inline PennMobile/src/main/java/com/pennapps/labs/pennmobile/adapters/DiningAdapter.java]: keep the else on the same line: ```} else {```
same on line 125

**Comment by dlakata** [comment]
LGTM after comments and squash

**Comment by adelq** [comment]
OK, no squashing I guess
screenshot 2016-02-08 at 7 49 51 pm

**Pull request merged**